app/models/user.rb

```diff
@@ -111,4 +95,13 @@ def admin?
   def chomp_username_attribute!
     self.username.chomp!
   end
+
+  # TokenAuthenticatable was removed from Devise in 3.1
+  # https://gist.github.com/josevalim/fb706b1e933ef01e4fb6
+  def generate_authentication_token
+    loop do
+      token = Devise.friendly_token
+      break token unless User.where(authentication_token: token).first
+    end
```

Comment:
Are we sure that this always breaks? I mean is there a risk of infinite loop?

Reply:
Copied from the Devise wiki. I trust that